### PR TITLE
Migrate pub/priv keys to use Secp256k1 

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

https://rchain.atlassian.net/browse/RCHAIN-3438

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by KentShikama** [comment]
bors try

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala [hunk 1]
```diff
@@ -977,7 +977,7 @@ class RegistryImpl[F[_]](
           val Some(Expr(GInt(_)))              = nonce.singleExpr
           // Then check the signature
           val Some(Expr(GByteArray(sigBytes))) = sig.singleExpr
-          if (keyBytes.size == Validator.Length && sigBytes.size == 71 &&
+          if (keyBytes.size == Validator.Length &&
```
Comment (tymm) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala]: Why not check if it is either 70 or 71?

File: casper/src/main/resources/PoS.rhox [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
+ 4.  | 1,         | secp256k1          | pk = 047b43d6548b72813b89ac1b9f9ca67624a8b372feedd71d4e2da036384a3e1236812227e524e6f237cde5f80dbb921cac12e6500791e9a9ed1254a745a816fe1f
+ 5.  | 4, 2,      | genIds             | uname = Unforgeable(0xa32c1649f1700469f125ea85f9c2a220072e593ae617213782d738e389639d3b)
+ 6.  | 3, 5,      | registry           | value = (9223372036854775807, bundle+ {   Unforgeable(0xa32c1649f1700469f125ea85f9c2a220072e593ae617213782d738e389639d3b) })
+ 7.  | 6,         | protobuf           | toSign = 2a40aa013d0a0d2a0b10feffffffffffffffff010a2c5a2a0a263a240a220a20a32c1649f1700469f125ea85f9c2a220072e593ae617213782d738e389639d3b1001
+ 8.  | 7, 1,      | secp256k1          | sig = 3044022054ff4bae3984252b116e41e28d98bb5533eaa39aec2729228159166e2784f641022066a0fd99e7ea33df812fab095cbe61250f9548bce6da3ec4c6a90c741b94087f
+ 9.  | 4,         | registry           | uri = rho:id:m3xk7h8r54dtqtwsrnxqzhe81baswey66nzw6m533nyd45ptyoybqr
+ ----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+new  PoS,
```
Comment (tymm) [inline casper/src/main/resources/PoS.rhox]: Formatting seems a little bit off here.

File: integration-tests/test/test_internal.py [hunk 3]
```diff
@@ -124,7 +124,7 @@ def test_parse_show_block_output() -> None:
...
...1} <- x7 ) {\n          @{x4}!((true, \"Bond successful!\")) |\n          @{Unforgeable(0x4de349932e7e2443526ad0a4923cd69c0027184659a6c475112fecca2008dedb)}!((x11).set(x0,((x2 - x8), x1, x3, ((x9).size() + 1))))\n        }\n      }\n    } |\n    for( @{x8}, @{x9}, @{x10}, @{x11} <= x6 ) {\n      match x8 {\n        {x13 : (x14, x15, x16, x17)...x12} => x6!(x12, x9, (x10).set(x13,((x14 + (x9).nth((x17 - 1))), x15, x16, x17)), x11) ;\n        _ => @{x11}!(x10)\n      }\n    }\n  }\n} |\nfor( @{x0}, @{x1}, @{x2} <= @{Unforgeable(0x39bdf789fb12e1c3487abce560193e3a9545d2d7e311aca70f6f47d36ba5115a)} ) {\n  new x3, x4, x5, x6, x7, x8 in {\n    @{bundle+ {   Unforgeable(0x6df7905a81f51e556695f3cbeccd1a81ffa1f3ab4c0884d18324d709cc8ea406) }}!(\"range\", 1, (x0 + 1), *x3) |\n    for( @{x9} <- x3 ) {\n      @{bundle+ {   Unforgeable(0x6df7905a81f51e556695f3cbeccd1a81ffa1f3ab4c0884d18324d709cc8ea406) }}!(\"map\", x9, *x4, *x5) |\n      for( @{x10} <- x5 ) {\n        @{bundle+ {   Unforgeable(0x6df7905a81f51e556695f3cbeccd1a81ffa1f3ab4c0884d18324d709cc8ea406) }}!(\"fold\", x10, 0, *x6, *x8) |\n        for( @{x11} <- x8 ) {\n          @{x2}!(([(((x10).nth(0) + x1) - x11)] ++ (x10).slice(1,x0)))\n
```
Comment (odeac) [inline integration-tests/test/test_internal.py]: can you explain the changes in this file?

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
bors try-

**Comment by KentShikama** [comment]
bors try

File: integration-tests/test/test_internal.py (hunk 3 above)
Comment (KentShikama) [reply integration-tests/test/test_internal.py]: These are internal tests for the integration test framework; the values shouldn't matter

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala (hunk 1 above)
Comment (KentShikama) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala]: It doesn't matter as the verify call will just fail if it has the wrong size

**Pull request merged**